### DISCUSSION TITLE
Re-enable OAuth password grant flow

### DISCUSSION THREAD

**Discussion by andregasser** [andregasser/bigbone#128] [General]
Hello @bocops, according to [this page](https://docs.joinmastodon.org/spec/oauth/), the OAuth password grant flow seems officially supported. I think we should re-enable it, even if there are risks involved when using it.

Also it makes writing integration tests much easier. as new user accounts can be registered with no browsers involved.

What do you think?

**Comment by bocops** [comment]
Oh, I haven't seen that one until now. Weird that this is not mentioned on the [oauth API methods](https://docs.joinmastodon.org/methods/oauth/) page, where two other `grant_type`s are explicitlcy mentioned.

We should definitely re-enable this if it is supported and especially if it is useful in testing, but should make it very obvious that this should not be used outside of tests and sample apps. Perhaps more or less revert #56, so that the result will be a deprecated and clearly documented method that does this?

**Reply by andregasser** [reply]
Agree. I will re-enable this as it was before. Created issue https://github.com/andregasser/bigbone/issues/132 for this.
